src/main/java/soolutions/hometaskapp/common/Amount.java

```diff
@@ -8,26 +8,36 @@
   private final String currency;
 
   public Amount(double value, String currency) {
-    if (value < 0) {
-      throw new InvalidAmountException(value);
-    }
-
     this.value = value;
     this.currency = currency;
   }
 
+  public Amount zero() {
+      return new Amount(0, this.currency);
+  }
+
   public Amount add(Amount other) {
     if (!currency.equals(other.currency)) {
       throw new CurrencyMismatchException(
-          "Cannot add different currencies " + currency + " : " + other.currency);
+          "Cannot apply different currencies " + currency + " : " + other.currency);
```

Comment:
If the method is called add, then the message should use the same terminology.

Reply:
I've moved this check into `ensureSameCurrency`, made it instance (to force first argument as `this`) and changed the message.